### PR TITLE
Fix index out of range panic

### PR DESCRIPTION
**Pull request by zacblazic**
If any passenger process that is position `n-1` in the list of processes is removed (usually due to it being killed), the very next call to `updateProcesses` will panic with an index out of range error.

```
panic: runtime error: index out of range [2] with length 2 [recovered]
	panic: runtime error: index out of range [2] with length 2

goroutine 53 [running]:
testing.tRunner.func1.1(0x151b6a0, 0xc00023a0a0)
	/usr/local/Cellar/go/1.14/libexec/src/testing/testing.go:941 +0x3d0
testing.tRunner.func1(0xc000536000)
	/usr/local/Cellar/go/1.14/libexec/src/testing/testing.go:944 +0x3f9
panic(0x151b6a0, 0xc00023a0a0)
	/usr/local/Cellar/go/1.14/libexec/src/runtime/panic.go:967 +0x15d
github.com/Intellection/passenger-exporter.updateProcesses(0xc000730270, 0xc00012e000, 0x2, 0x2, 0x3, 0xc000730401)
	/Users/zacblazic/workspace/go/src/github.com/Intellection/passenger-exporter/passenger_exporter.go:378 +0x493
github.com/Intellection/passenger-exporter.TestUpdateProcessIdentifiers(0xc000536000)
```

The two factors at play here are:
* The length of the `found` array.
* The index values of the remaning processes.

If we have 3 processes indexed as follows:
* 0 - `p1`
* 1 - `p2`
* 2 - `p3`

Assume the first process is killed, and we then call `updateProcesses`, we get the following output:
* 1 - `p2`
* 2 - `p3`

Time has passed and we call `updateProcesses` once again. When we do so, the `old` process list is as follows:
* 1 - `p2`
* 2 - `p3`

Note that the `old` process list has a length of 2 but contains an entry with an index that is larger than the length of the map. When the `found` array is accessed with the `p3` index, we get the index out of range error.

**TLDR;** If you remove `p1` or `p2`, you still have `p3` with an index of 2, yet the length of the `found` array is 2.

### CODE REVIEW

File: passenger_exporter_test.go [hunk 1]
```diff
@@ -158,6 +162,7 @@ func TestUpdateProcessIdentifiers(t *testing.T) {
 				Process{PID: "cdf"},
 				Process{PID: "dfe"},
 			},
+			6,
```
Comment (itskingori) [inline passenger_exporter_test.go:165]: How have you come up with `6` here?

File: passenger_exporter_test.go [hunk 2]
```diff
@@ -174,6 +179,7 @@ func TestUpdateProcessIdentifiers(t *testing.T) {
 				Process{PID: "jkl"},
 				Process{PID: "lmn"},
 			},
+			9,
```
Comment (itskingori) [inline passenger_exporter_test.go:182]: How have you come up with `9` here?

File: passenger_exporter_test.go [hunk 3]
```diff
@@ -224,6 +257,7 @@ func TestInsertingNewProcesses(t *testing.T) {
 			Process{PID: "newPID"},
 			Process{PID: "newPID2"},
 		},
+		6,
```
Comment (itskingori) [inline passenger_exporter_test.go:260]: How have you come up with `6` here?

**Review by itskingori — COMMENTED** [review]
I have a high level idea of what's going on there largely because of your explanation and also seeing that we're rebuilding the map using `maxProcesses` (minus the ones that have been killed) instead of `len(old)`, so the index is now correct.

In the code we're getting the new count from `info.MaxProcessCount` but in the tests I don't quite understand how you came up with the values. 🤔

See inline comments for the one's that are puzzling me the most.

File: passenger_exporter_test.go (hunk 1 above)
Comment (zacblazic) [reply passenger_exporter_test.go:165]: Max processes is controlled by the [`--max-pool-size`](https://www.phusionpassenger.com/library/config/standalone/reference/#--max-pool-size-max_pool_size) setting. By default its value is 6. The idea is that this value will always be greater or equal to the number of processes at any given point, hence making it a good value for setting the size of the `found` array.

The values chosen in tests are arbitrary, but always greater or equal to the number of items in the `input` map and `processes` array of `updateProcessSpec`.

File: passenger_exporter_test.go (hunk 2 above)
Comment (zacblazic) [reply passenger_exporter_test.go:182]: See https://github.com/Intellection/passenger-exporter/pull/14#discussion_r394157681.

File: passenger_exporter_test.go (hunk 3 above)
Comment (zacblazic) [reply passenger_exporter_test.go:260]: See https://github.com/Intellection/passenger-exporter/pull/14#discussion_r394157681.

File: passenger_exporter_test.go (hunk 1 above)
Comment (itskingori) [reply passenger_exporter_test.go:165]: 👌

Comment (itskingori) [reply passenger_exporter_test.go:165]: Ok, that explains why in the code we're checking `if pid == "" {`:

https://github.com/Intellection/passenger-exporter/blob/67ec97b6b8cd953fe9cd54b7cf7256abeef0c08e/passenger_exporter.go#L394-L395

Comment (zacblazic) [reply passenger_exporter_test.go:165]: For interest sake, it panics on line 377:

https://github.com/Intellection/passenger-exporter/blob/67ec97b6b8cd953fe9cd54b7cf7256abeef0c08e/passenger_exporter.go#L375-L377

**Review by itskingori — APPROVED** [review]
